diff -Naur libsodium-1.0.17/dist-build/emscripten.sh libsodium-1.0.17-new/dist-build/emscripten.sh
--- libsodium-1.0.17/dist-build/emscripten.sh	2019-01-07 07:29:47.000000000
+++ libsodium-1.0.17-new/dist-build/emscripten.sh	2019-02-19 22:35:24.000000000
@@ -64,7 +64,7 @@
 emconfigure ./configure $CONFIG_EXTRA --disable-shared --prefix="$PREFIX" \
                         --without-pthreads \
                         --disable-ssp --disable-asm --disable-pie \
-                        CFLAGS="$CFLAGS" && \
+                        CFLAGS="$CFLAGS -s WASM_OBJECT_FILES=1" && \
 emmake make clean
 [ $? = 0 ] || exit 1
 
@@ -76,6 +76,7 @@
       "${PREFIX}/lib/libsodium.a" -o "${outFile}" || exit 1
   }
   emmake make $MAKE_FLAGS install || exit 1
+  exit 0
   emccLibsodium "${PREFIX}/lib/libsodium.asm.tmp.js" -Oz -s WASM=0 -s RUNNING_JS_OPTS=1
   emccLibsodium "${PREFIX}/lib/libsodium.wasm.tmp.js" -O3 -s WASM=1
 
diff -Naur libsodium-1.0.17/src/libsodium/randombytes/randombytes.c libsodium-1.0.17-new/src/libsodium/randombytes/randombytes.c
--- libsodium-1.0.17/src/libsodium/randombytes/randombytes.c	2018-09-08 09:54:55.000000000
+++ libsodium-1.0.17-new/src/libsodium/randombytes/randombytes.c	2019-02-19 22:36:39.000000000
@@ -69,6 +69,8 @@
 #endif
 }
 
+extern uint32_t js_random_uint32();
+
 uint32_t
 randombytes_random(void)
 {
@@ -76,9 +78,7 @@
     randombytes_init_if_needed();
     return implementation->random();
 #else
-    return EM_ASM_INT_V({
-        return Module.getRandomValue();
-    });
+    return js_random_uint32();
 #endif
 }
 
